### PR TITLE
fix #13

### LINKED ISSUE #13: ColorPickerの色を、JS側から追加するように変更する
- [ ] draw-tools.js 内、ColorPicker.prototype.addPalette の内容を書く
- [ ] HTMLからli要素を消す
- [ ] 新しく追加した要素に、listenerを追加するのを忘れない！！！！！（ぼくだと忘れる）

### PR DESCRIPTION
**Pull request by Ranats**
This change is [Reviewable](https://reviewable.io/reviews/comozilla/parapara-canvas-editor/42)

### CODE REVIEW

**Comment by shundroid** [comment]
Reviewed 2 of 4 files at r1.
Review status: 2 of 4 files reviewed at latest revision, 1 unresolved discussion.

---

_[js/color-picker.js, line 10 [r1]](https://reviewable.io:443/reviews/comozilla/parapara-canvas-editor/42#-KG1HiJuelkS4_C15QO5:-KG1HiJvbDx9WofRa3-5:1206012686) ([raw file](https://github.com/comozilla/parapara-canvas-editor/blob/24acadb344c7e8cbdb8b57805ab4904e6c44fb9d/js/color-picker.js#L10)):_
すみません。ここのコメントを削除していただけませんでしょうか。
よろしくお願いします。

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/parapara-canvas-editor/42)_

**Comment by shundroid** [comment]
Review status: 2 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

_[js/color-picker.js, line 16 [r1]](https://reviewable.io:443/reviews/comozilla/parapara-canvas-editor/42#-KG1HoQh3BHjBqiCmDkq:-KG1HoQh3BHjBqiCmDkr:1758454935) ([raw file](https://github.com/comozilla/parapara-canvas-editor/blob/24acadb344c7e8cbdb8b57805ab4904e6c44fb9d/js/color-picker.js#L16)):_
ここの変数の名前ですが、objより、
pallete などのほうが良さそうです。

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/parapara-canvas-editor/42)_

**Comment by shundroid** [comment]
Reviewed 1 of 4 files at r1.
Review status: 3 of 4 files reviewed at latest revision, 2 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/parapara-canvas-editor/42)_

File: js/color-picker.js [hunk 1]
```diff
@@ -2,16 +2,35 @@ function ColorPicker(elem, defaultColor) {
...
+  obj.addEventListener("click", event => {
+    this.color = event.target.style.backgroundColor;
+  });
+
 };
+
+function isColor(color) {
+  var testElement = document.createElement("span");
+  var beforeColor = testElement.style.backgroundColor =
+      color === "red" ? "green" : "red";
```
Comment (dadaa) [inline js/color-picker.js]: なんら background-color の指定のない element の style.backgroundColor は空文字なので，単純に値を testElement.style.backgroundColor に設定し，これが変わっていれば正しい色と言えそうです．

File: js/main.js [hunk 2]
```diff
@@ -16,10 +16,19 @@ document.addEventListener("DOMContentLoaded", function() {
...
+      ["red", "orange","yellow","lightgreen",
+        "green","skyblue","purple","black","white"];
+
+  colorList.forEach(color => {
+    colorPicker.addPalette(color);
+  });
+
   lineWidthPicker =
     new LineWidthPicker(document.getElementById("menu-line-width"),
                                   10);
```
Comment (dadaa) [inline js/main.js]: ちょっと違うところだけど，不思議な改行だねw

File: js/color-picker.js (hunk 1 above)
Comment (dadaa) [inline js/color-picker.js]: これ．そのように直してもらってもよいでしょうか？？

Comment (dadaa) [inline js/color-picker.js]: これ，そのように直していただいてもよいでしょうか？？

File: js/main.js [hunk 3]
```diff
@@ -16,13 +16,22 @@ document.addEventListener("DOMContentLoaded", function() {
...
+        "green","skyblue","purple","black","white"];
+
+  colorList.forEach(color => {
+    colorPicker.addPalette(color);
+  });
+
   lineWidthPicker =
     new LineWidthPicker(document.getElementById("menu-line-width"),
-                                  10);
+        10);
```
Comment (dadaa) [inline js/main.js]: なんか，格好悪いすねw
シンプルに，

``` js
lineWidthPicker = 
  new LineWidthPicker(document.getElementById("menu-line-width"), 10);
```

でいかがでしょう？

File: js/main.js [hunk 4]
```diff
@@ -16,13 +16,21 @@ document.addEventListener("DOMContentLoaded", function() {
   setListenerForCanvas(firstFrameId);
   framesController.setCurrentFrame(firstFrameId);
 
+
```
Comment (dadaa) [inline js/main.js]: この行，いらないかも

File: js/main.js [hunk 5]
```diff
@@ -16,13 +16,21 @@ document.addEventListener("DOMContentLoaded", function() {
...
   colorPicker = new ColorPicker(
     document.getElementById("menu-colors"),
     "red");
 
+  var colorList =
+      ["red", "orange","yellow","lightgreen",
+        "green","skyblue","blue","purple","black","white"];
```
Comment (dadaa) [inline js/main.js]: インデント，細かいけど，こんな感じで，左のダブルコーテーションを合わせましょう!
["red", "orange","yellow","lightgreen",
 "green","skyblue","blue","purple","black","white"];

**Comment by dadaa** [comment]
Reviewed 1 of 2 files at r4, 1 of 1 files at r5.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/parapara-canvas-editor/42)_

**Comment by dadaa** [comment]
Reviewed 2 of 4 files at r1.
Review status: all files reviewed at latest revision, 7 unresolved discussions.

---

_Comments from [Reviewable](https://reviewable.io:443/reviews/comozilla/parapara-canvas-editor/42)_

**Comment by dadaa** [comment]
おつかれさまでしたー

**Pull request merged**